utils/upgrade-prompts.js

```diff
@@ -0,0 +1,96 @@
+// @flow
+const inquirer = require('inquirer');
+const {rsort} = require('semver');
+
+/*::
+type PromptForTypesVersion = (
+  string,
+  ?string,
+  Array<string>,
+  boolean
+) => Promise<?string>;
+*/
+
+const createPromptChoices = latest => [
+  {name: `Use latest version (${latest})`, value: 'latest'},
+  {name: 'Enter a specific version manually', value: 'manual'},
+  {name: 'Skip this package', value: 'skip'},
+  {name: 'Abort the upgrade process', value: 'abort'},
+];
+
+const promptForManualVersion = async versions => {
+  const {manualVersion} = await inquirer.prompt([
+    {
+      type: 'input',
+      name: 'manualVersion',
+      message: 'Enter the specific version:',
+      validate: input => {
+        if (!input.trim()) return 'Version cannot be empty';
+        if (!versions.includes(input.trim())) {
+          return `Version "${input.trim()}" not found. Available versions: ${versions.join(
+            ', '
+          )}`;
+        }
+        return true;
+      },
+    },
+  ]);
+  return manualVersion.trim();
+};
+
+const handlePromptAction = async (action, latest, versions) => {
+  switch (action) {
+    case 'latest':
+      return latest;
+    case 'manual':
+      return await promptForManualVersion(versions);
+    case 'skip':
+      return null;
+    case 'abort':
+      throw new Error('Upgrade process aborted by user');
+    default:
+      return null;
+  }
+};
+
+const promptForTypesVersion /*: PromptForTypesVersion */ = async (
+  typesPackageName,
+  originalRange,
+  versions,
+  interactive = true
+) => {
+  // Sort versions to find the actual latest (highest) version
+  const sortedVersions = rsort(versions);
```

Comment:
Assuming this "just works" since it's straight from the semver dependency, but worth noting that we sometimes have packages that do funky things like release a bunch of canary versions:

```
  "19.3.0-canary-93f85932-20251016",
  "19.3.0-canary-9724e3e6-20251008",
  "19.3.0-canary-a4eb2dfa-20251006",
  ...
```

These might technically be "greater versions" but we don't necessarily want folks upgrading to them.